uint8 state_code
string state_description